src/runtime_src/core/tools/xbutil2/SubCmdExamine.h

```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright (C) 2020 Xilinx, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+#ifndef __SubCmdExamine_h_
+#define __SubCmdExamine_h_
+
+#include "tools/common/SubCmd.h"
+
+class SubCmdExamine : public SubCmd {
+ public:
+   virtual void execute(const SubCmdOptions &_options) const;
+
+ public:
+   SubCmdExamine(bool _isHidden, bool _isDepricated, bool _isPreliminary);
+   virtual ~SubCmdExamine() {};
+
+ // Methods not supported
+ private:
```

Comment:
Lets simplify all derived class and remove the explicit delete of these functions which will not be created by the compiler anyway.   You can keep the deleted assignment in the base class, but the ctors don't matter because the is already a non default ctor defined.

The destructor also doesn't have any effect since the default one will do just fine.

Reply:
I understand where you are coming from and it makes sense to remove this destructor, which I did.  

That said, I don't like it when the the "compiler" creates methods on the fly with sometimes unexpected behavior, hence why I originally added it (a habit to do so).  And yes, it does reading the header file easier as well.